### PR TITLE
Pagination added to Source for Regulatory effects to decrease load ti…

### PR DESCRIPTION
**Pull request by UAMS16**
…me and  Created SRE html  page to display table.

### CODE REVIEW

**Review by tncowart — CHANGES_REQUESTED** [review]
This is looking great! I've left a few comments on the code. 

Additionally, this PR should include tests. the json template should have some tests -- you can probably mostly copy `json_templates/v1/tests/test_reg_effect.py` for that. You should also include an "endpoint" test, to make sure the endpoint is working for both HTML and JSON requests. See `views/v1/tests/test_dna_features.py` and `views/v1/tests/test_experiment.py` for an idea of what these look like.

File: cegs_portal/search/views/v1/reg_effects.py [hunk 1]
```diff
@@ -53,38 +59,13 @@ def request_options(self, request):
         """
         options = super().request_options(request)
         options["page"] = int(request.GET.get("page", 1))
-
+        options["per_page"] = int(request.GET.get("per_page", 20))
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/views/v1/reg_effects.py:62]: Replaced Line 67 with reg_effect_paginator = Paginator(reg_effects, options["per_page"])

File: cegs_portal/search/views/v1/reg_effects.py [hunk 2]
```diff
@@ -3,9 +3,14 @@
 from cegs_portal.search.json_templates.v1.reg_effect import (
     regulatory_effect,
     source_reg_effects,
+
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/views/v1/reg_effects.py:6]: 1.	Empty line removed
2.	Deleted source_reg_effects in
cegs_portal.search.json_templates.v1.reg_effect
3.	Now using source_reg_effects in 
cegs_portal.search.json_templates.v1.source_reg_effects

File: cegs_portal/search/view_models/v1/reg_effects.py [hunk 3]
```diff
@@ -26,10 +26,11 @@ def source_search(cls, source_id: str):
             RegulatoryEffectObservation.objects.with_facet_values()
             .filter(sources__accession_id=source_id)
             .prefetch_related(
-                "experiment__biosamples__cell_line",
-                "sources__source_for",
-                "targets__target_of__sources",
+                "experiment",
```
Comment (tncowart) [inline cegs_portal/search/view_models/v1/reg_effects.py:29]: 👍

File: cegs_portal/search/views/v1/reg_effects.py [hunk 4]
```diff
@@ -1,11 +1,15 @@
 from django.core.paginator import Paginator
 
-from cegs_portal.search.json_templates.v1.reg_effect import (
+from cegs_portal.search.json_templates.v1.source_reg_effects import (
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:3]: I think that `source_reg_effects` should come from `cegs_portal.search.json_templates.v1.source_reg_effects` but that `regulatory_effect` should come from `cegs_portal.search.json_templates.v1.reg_effect`

**Review by tncowart — CHANGES_REQUESTED** [review]

File: cegs_portal/search/conftest.py [hunk 5]
```diff
@@ -50,10 +50,8 @@ def paged_experiments() -> Pageable[Experiment]:
     e6 = ExperimentFactory(other_files=(other_file(), other_file()), biosamples=(BiosampleFactory(),))
     e6.data_files.set(((data_file(e6),)))
     experiments = sorted([e1, e2, e3, e4, e5, e6], key=lambda x: x.accession_id)
-    return MockPaginator(
-        experiments,
-        3,
-    )
+    pages = MockPaginator(experiments, 3)
+    return pages.page(1)
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/conftest.py:54]: I think you changed it that Friday we went over tests. I have not worked with Experiment Tests.

[the remark this answers is not in the document]
Comment (tncowart) [reply cegs_portal/search/conftest.py:54]: Ah, funny. I don't remember that at all!

[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/conftest.py:54]: Ha! Yes, you were going pretty fast editing code etc. You usually just quickly fix or change things along the way while we go though code. Its like picking up trash along the road--nobody remembers when they do it.

**Pull request merged**